src/Raven.Server/Documents/Replication/OutgoingReplicationHandler.cs

```diff
@@ -118,15 +120,17 @@ public string GetNode()
         private void ReplicateToDestination()
         {
             AddReplicationPulse(ReplicationPulseDirection.OutgoingInitiate);
-
             NativeMemory.EnsureRegistered();
             try
             {
-                var connectionInfo = ReplicationUtils.GetTcpInfo(Destination.Url, GetNode(), "Replication",
-                    _parent._server.Server.ClusterCertificateHolder.Certificate);
+                if (ConnectionInfo == null)
```

Comment:
This is wrong, we need to ensure that whenever we call `ReplicateToDestination`  we'll always renew the check for the TCP details.